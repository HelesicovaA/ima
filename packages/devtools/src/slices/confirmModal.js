import { createSlice } from 'redux-starter-kit';

export const confirmModalInitialState = {
  body: '',
  opened: false,
  accept: null,
  cancel: null
};

const confirmModal = createSlice({
  slice: 'confirmModal',
  initialState: confirmModalInitialState,
  reducers: {
    showConfirmModal(state, { payload: { body, accept, cancel } }) {
      state.opened = true;
      state.body = body || confirmModalInitialState.body;
      state.accept = accept || confirmModalInitialState.accept;
      state.cancel = cancel || confirmModalInitialState.cancel;
    },
    hideConfirmModal(state) {
      state.opened = false;
    }
  }
});

const { reducer, actions } = confirmModal;

export { reducer, actions };
