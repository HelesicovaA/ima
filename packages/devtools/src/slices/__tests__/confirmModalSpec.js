import { confirmModalInitialState, reducer, actions } from '../confirmModal';

describe('confirmModalInitialState', () => {
  it('should match snapshot', () => {
    expect(confirmModalInitialState).toMatchSnapshot();
  });
});

describe('reducer', () => {
  let curState;

  beforeEach(() => {
    curState = {
      body: '',
      opened: false,
      accept: null,
      cancel: null
    };
  });

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(confirmModalInitialState);
  });

  it('should set opened to false', () => {
    expect(
      reducer(curState, { type: 'confirmModal/hideConfirmModal' })
    ).toEqual({
      ...curState,
      opened: false
    });
  });

  it('should set opened to true and set data to state', () => {
    expect(
      reducer(curState, {
        type: 'confirmModal/showConfirmModal',
        payload: {
          body: 'body',
          accept: 'accept',
          cancel: 'cancel'
        }
      })
    ).toEqual({
      opened: true,
      body: 'body',
      accept: 'accept',
      cancel: 'cancel'
    });
  });

  it('should set opened to true and data to initial state if not provided', () => {
    expect(
      reducer(curState, {
        type: 'confirmModal/showConfirmModal',
        payload: {}
      })
    ).toEqual({
      opened: true,
      body: confirmModalInitialState.body,
      accept: confirmModalInitialState.accept,
      cancel: confirmModalInitialState.cancel
    });
  });
});

describe('actions', () => {
  it('should create action to hide confirm modal', () => {
    expect(actions.hideConfirmModal()).toEqual({
      type: 'confirmModal/hideConfirmModal'
    });
  });

  it('should create action to show confirm modal', () => {
    const data = {};

    expect(actions.showConfirmModal(data)).toEqual({
      type: 'confirmModal/showConfirmModal',
      payload: {}
    });
  });
});
