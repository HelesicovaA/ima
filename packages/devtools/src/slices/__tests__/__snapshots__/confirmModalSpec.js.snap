// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirmModalInitialState should match snapshot 1`] = `
Object {
  "accept": null,
  "body": "",
  "cancel": null,
  "opened": false,
}
`;
